examples/energy-management-app/energy-management-common/include/EVSEManufacturerImpl.h

```diff
@@ -33,22 +33,37 @@ namespace EnergyEvse {
 class EVSEManufacturer
 {
 public:
+    EVSEManufacturer(EnergyEvseManager * aInstance) { mInstance = aInstance; }
+    EnergyEvseManager * GetInstance() { return mInstance; }
+    EnergyEvseDelegate * GetDelegate()
+    {
+        if (mInstance)
+        {
+            return mInstance->GetDelegate();
+        }
+        return nullptr;
+    }
+
     /**
      * @brief   Called at start up to apply hardware settings
      */
-    CHIP_ERROR Init(EnergyEvseManager * aInstance);
+    CHIP_ERROR Init();
 
     /**
      * @brief   Called at shutdown
      */
-    CHIP_ERROR Shutdown(EnergyEvseManager * aInstance);
+    CHIP_ERROR Shutdown();
 
     /**
      * @brief   Main Callback handler from delegate to user code
      */
     static void ApplicationCallbackHandler(const EVSECbInfo * cb, intptr_t arg);
 
 private:
+    EnergyEvseManager * mInstance;
+
+    int64_t mLastChargingEnergyMeter    = 0;
+    int64_t mLastDischargingEnergyMeter = 0;
```

Comment:
In Matter all of the energy units are mWh - and this is why we had to use int64_t everywhere.
Maybe we could create a new typedef (like percent, epoch_s).

When we did the ZAP changes to add the autogen from the XML it just converted them to int64_t.
I wasn't sure how to add a new Typedef like energy_mwh types (such as percent).

Slightly reluctant to rename all the variables since we've agreed that these will always be in mW, mV, mWh, mA.
and I've tried to be consistent with the attribute names (based on the names in the specs - which don't include the units).

Whilst these in this case aren't attributes they are just copies of the attribute values.
If it does help I can change the variable names here.